src/components/tx-flow/flows/NewSpendingLimit/CreateSpendingLimit.tsx

```diff
@@ -46,15 +36,22 @@ export const _validateSpendingLimit = (val: string, decimals?: number) => {
   }
 }
 
-export const SpendingLimitForm = ({ data, onSubmit }: Props) => {
+export const CreateSpendingLimit = ({
+  params,
+  onSubmit,
+}: {
+  params: NewSpendingLimitFlowProps
+  onSubmit: (data: NewSpendingLimitFlowProps) => void
+}) => {
   const chainId = useChainId()
   const [showResetTime, setShowResetTime] = useState<boolean>(false)
   const { balances } = useVisibleBalances()
 
   const resetTimeOptions = useMemo(() => getResetTimeOptions(chainId), [chainId])
+  const defaultResetTime = resetTimeOptions[0].value
 
-  const formMethods = useForm<NewSpendingLimitData>({
-    defaultValues: { ...data, resetTime: '0' },
+  const formMethods = useForm<NewSpendingLimitFlowProps>({
+    defaultValues: params,
```

Comment:
Somehow when going back to this step the tokenAddress resets to undefined and I am getting warnings in the console

Reply:
Seems like there's a bug with the `register` function. I've fixed it and added a comment to look into this further when we update react-hook-form.